extras/kms/kms.go

```diff
@@ -205,19 +195,29 @@ func (k *Kms) GetWrapper(ctx context.Context, scopeId string, purpose KeyPurpose
 	// Fast-path: we have a valid key at the scope/purpose. Verify the key with
 	// that ID is in the multiwrapper; if not, fall through to reload from the
 	// DB.
-	val, ok := k.scopedWrapperCache.Load(scopedPurpose(scopeId, purpose))
-	if ok {
-		wrapper, ok := val.(*multi.PooledWrapper)
-		if !ok {
-			return nil, fmt.Errorf("%s: scoped wrapper is not a multi.PooledWrapper: %w", op, ErrInternal)
-		}
-		if opts.withKeyId == "" {
-			return wrapper, nil
-		}
-		if keyIdWrapper := wrapper.WrapperForKeyId(opts.withKeyId); keyIdWrapper != nil {
-			return keyIdWrapper, nil
+	currVersion, err := currentCollectionVersion(ctx, k.repo.reader)
+	if err != nil {
+		return nil, fmt.Errorf("%s: unable to determine current version of the kms collection: %w", op, err)
+	}
+	switch {
+	case currVersion > atomic.LoadUint64(&k.collectionVersion):
+		k.clearCache(ctx)
+		atomic.StoreUint64(&k.collectionVersion, currVersion)
+	default:
+		val, ok := k.scopedWrapperCache.Load(scopedPurpose(scopeId, purpose))
+		if ok {
+			wrapper, ok := val.(*multi.PooledWrapper)
+			if !ok {
+				return nil, fmt.Errorf("%s: scoped wrapper is not a multi.PooledWrapper: %w", op, ErrInternal)
+			}
+			if opts.withKeyId == "" {
+				return wrapper, nil
+			}
+			if keyIdWrapper := wrapper.WrapperForKeyId(opts.withKeyId); keyIdWrapper != nil {
+				return keyIdWrapper, nil
+			}
+			// Fall through to refresh our multiwrapper for this scope/purpose from the DB
```

Comment:
Is this the behavior we want if `currVersion < atomic.LoadUint64(&k.collectionVersion)`?

Reply:
Yea, I think it might be possible if `currVersion` eventually rolls over, which admittedly shouldn't happen for a very very long time given that it we are using `bigint` for postgres, and it looks like `int` for sqlite has the same max value. But I think if it did to happen, we wouldn't want the behavior the `default` case.

Reply:
agreed. ty!